src/gurt/telemetry.c

```diff
@@ -0,0 +1,1857 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+/*
+ * telemetry: TELEMETRY common logic
+ */
+#define D_LOGFAC	DD_FAC(telem)
+
+#include <gurt/common.h>
+
+#include <string.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <time.h>
+#include <inttypes.h>
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
+#include <sys/syscall.h>
+#include <sys/mman.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <pthread.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <gurt/telemetry.h>
+
+/*
+ * These are server side global variables that apply to the entire server
+ * side process
+ */
+struct d_tm_node_t *root;
+uint64_t *shmemRoot;
+uint8_t *shmemIdx;
+uint64_t shmemFree;
+pthread_mutex_t addlock;
```

Comment:
Given that these are global variables, it seems like they should have names that more clearly identify them as being part of this API. Might be worth looking in the rest of the codebase for style precedent, or consulting with some of the core DAOS developers for guidance.

Reply:
Sure, I can give them better names.

Reply:
Done